zerver/tests/test_queue.py

```diff
@@ -0,0 +1,16 @@
+import mock
+import os
+from typing import Any
+import ujson
+
+from pika.exceptions import ConnectionClosed
+
+from zerver.lib.queue import TornadoQueueClient
+from zerver.lib.test_classes import ZulipTestCase
+
+class TestTornadoQueueClient(ZulipTestCase):
+    @mock.patch('zerver.lib.queue.ExceptionFreeTornadoConnection', autospec=True)
+    def test_on_open_closed(self, mock_cxn):
+        connection = TornadoQueueClient()
+        connection.connection.channel.side_effect = ConnectionClosed
+        connection._on_open(mock.MagicMock())
```

Comment:
Since we're trying to avoid making test output any less clean, we should probably mock `log.warning` here.

Reply:
Oh ha, thanks.

I'd originally sketched that exception handler as just `pass`, and I think I didn't rerun the test after giving it a proper warning. >_>

I'm increasingly coming to think that we should solve this problem in a single place; lots of our test code is made noisier to read by mocking `logging.warning` and `logging.error` and various logger objects and so on, and the code that isn't often should be because it's noisy to run, like this. That's a problem for another day, though.

Reply:
I guess my perspective is that it's valuable for the test code to make clear (and verify) which things should produce log output vs. not.  What we're missing right now is an automated check for new log output being produced that would prevent regressions here, which there's an open issue for.